pkg/stanza/fileconsumer/util_test.go

```diff
@@ -84,13 +85,28 @@ func withEmitChan(emitChan chan *emitParams) testManagerOption {
 	}
 }
 
-func buildTestManager(t *testing.T, cfg *Config, opts ...testManagerOption) (*Manager, chan *emitParams) {
+func withReaderChan() testManagerOption {
+	return func(c *testManagerConfig) {
+		c.initializeChannel = true
+	}
+}
+
+func buildTestManager(t testing.TB, cfg *Config, opts ...testManagerOption) (*Manager, chan *emitParams) {
 	tmc := &testManagerConfig{emitChan: make(chan *emitParams, 100)}
 	for _, opt := range opts {
 		opt(tmc)
 	}
 	input, err := cfg.Build(testutil.Logger(t), testEmitFunc(tmc.emitChan))
 	require.NoError(t, err)
+	if tmc.initializeChannel && useThreadPool.IsEnabled() {
```

Comment:
We probably don't need to check the feature gate in tests, right? We're in full control via the `withReaderChan` option.

Reply:
Actually, we do need this check here or at [line#90](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/25884/files#diff-540e11ad5bede83a0150f6742d29bca27572279b92fb606af37527d5159837ceR90).
Because we don't place a check before calling `withReaderChan` in test cases. That would result in hoge redundant code.

Reply:
If we have code here that checks the feature gate, then do we need the `withReaderChan` option at all? It seems redundant that we're passing `withReaderChan` to all these tests, but then only using it if the feature gate is enabled. Can we just set up and use the channel based on the feature gate alone?

Reply:
Now this is a tricky situation.
Let me give a try to another approach I'm currently thinking.